lib/src/Uncertainty/Distribution/CombinationsDistribution.cxx

```diff
@@ -0,0 +1,521 @@
+  }
+  if (dimension > 1)
+  {
+    PointWithDescription point(getParameter());
+    point.setDescription(getParameterDescription());
+    point.setName("dependence");
+    parameters[dimension] = point;
+  }
+  return parameters;
+}
+
+/* Parameters value accessor */
+Point CombinationsDistribution::getParameter() const
+{
+  return {static_cast<Scalar>(k_), static_cast<Scalar>(n_)};
+}
+
+void CombinationsDistribution::setParameter(const Point & parameter)
+{
+  if (parameter.getSize() != 2) throw InvalidArgumentException(HERE) << "CombinationsDistribution expected 2 parameters, got " << parameter.getSize();
+  if (parameter[0] != std::round(parameter[0])) throw InvalidArgumentException(HERE) << "CombinationsDistribution first parameter k must be an integer, got " << parameter[0];
+  if (parameter[1] != std::round(parameter[1])) throw InvalidArgumentException(HERE) << "CombinationsDistribution second parameter n must be an integer, got " << parameter[1];
+  const Scalar w = getWeight();
+  *this = CombinationsDistribution(parameter[0], parameter[1]);
+  setWeight(w);
+}
+
+/* Parameters description accessor */
+Description CombinationsDistribution::getParameterDescription() const
+{
+  return {"k", "n"};
+}
+
+void CombinationsDistribution::update()
+{
+  logPDFValue_ = -1.0 * SpecFunc::LogBinomialCoefficient(n_, k_);
+  isAlreadyComputedMean_ = false;
+  isAlreadyComputedCovariance_ = false;
+  isAlreadyCreatedGeneratingFunction_ = false;
+  computeRange();
+}
+
+/* K accessor */
+void CombinationsDistribution::setK(const UnsignedInteger k)
+{
+  if (k == 0) throw InvalidArgumentException(HERE) << "Error: k must be > 0.";
+  if (k > n_) throw InvalidArgumentException(HERE) << "Error: k must be less or equal to n, here k=" << k << " and n=" << n_;
+  if (k != k_)
+  {
+    k_ = k;
+    setDimension(k);
+    update();
+  }
+}
+
+/* K accessor */
+UnsignedInteger CombinationsDistribution::getK() const
+{
+  return k_;
+}
+
+/* N accessor */
+void CombinationsDistribution::setN(const UnsignedInteger n)
+{
+  if (n == 0) throw InvalidArgumentException(HERE) << "Error: n must be > 0.";
+  if (n < k_) throw InvalidArgumentException(HERE) << "Error: n must be greater or equal to k, here n=" << n << " and k=" << k_;
+  if (n != n_)
+  {
+    n_ = n;
+    update();
+  }
+}
+
+UnsignedInteger CombinationsDistribution::getN() const
+{
+  return n_;
+}
+
+/* K/N accessor */
+void CombinationsDistribution::setKN(const UnsignedInteger k,
+                                      const UnsignedInteger n)
+{
+  if (k == 0) throw InvalidArgumentException(HERE) << "Error: k must be > 0.";
+  if (n == 0) throw InvalidArgumentException(HERE) << "Error: n must be > 0.";
+  if (k > n) throw InvalidArgumentException(HERE) << "Error: k must be less or equal to n, here k=" << k << " and n=" << n;
+  k_ = k;
+  setDimension(k);
+  n_ = n;
+  logPDFValue_ = SpecFunc::LogGamma(n_ - k_ + 1) - SpecFunc::LogGamma(n_ + 1) + SpecFunc::LogGamma(k_ + 1);
```

Comment:
Except for isAlreadyGeneratedGenaratingFunction, this is the body of update isnt'it?